client/src/model/component/CircuitDiagram.hx

```diff
@@ -215,4 +215,124 @@ class CircuitDiagram extends Observable implements CircuitDiagramI implements Ob
+					if (connection.aPortIsConnecte() && otherConnection.aPortIsConnecte()) {
+						this.addLink(new Link(this, connection.get_xPosition(), connection.get_yPosition(), otherConnection.get_xPosition(), otherConnection.get_yPosition()));
+					} else {
+						if (connection.aPortIsConnecte()) {
+							for (element in connection.get_connectedElements()) {
+								otherConnection.connect(Std.downcast(element, Connectable));
+							}
+							markedConnections.push(connection);
+						}
+					}
+					
+					
+				} 
+			}
+		}
+		
+		// removing marked connections. Unclear is swapping connections in use cases is actually what it is supposed to be.
+		// removing such unused connections makes more sense. CHECK ?
+		for (connection in markedConnections) {
+			connectionArray.remove(connection);
+		}
+		
+		// For all connections c and for all links x not connected to c
+		// 		If c is not close to either endpoint of x, but is close to x
+        // 			Split x into two links each with an endpoint added to c
+		// VERIFY :: Can cause Short circuit!!
+		// this carries on with the remaining links
+		var removeLinks = new Array<Link>();
+		for (connection in connectionArray) {
+			for (link in this.linkArray) {
+				if ((link.get_endpoint(0).getConnection() != connection) && (link.get_endpoint(1).getConnection() != connection)) {
+					if (Link.pointToLine(connection.get_xPosition(), connection.get_yPosition(), link.get_endpoint(0).get_xPosition(), link.get_endpoint(0).get_yPosition(), link.get_endpoint(1).get_xPosition(), link.get_endpoint(1).get_yPosition()) <= 15) {
+						var link1 = new Link(this, link.get_endpoint(0).get_xPosition(), link.get_endpoint(0).get_yPosition(), connection.get_xPosition(), connection.get_yPosition());
+						this.addLink(link1);
+						connection.connect(link1.get_endpoint(1));
+						var link2 = new Link(this, connection.get_xPosition(), connection.get_yPosition(), link.get_endpoint(1).get_xPosition(), link.get_endpoint(1).get_yPosition());
+						this.addLink(link2);
+						connection.connect(link2.get_endpoint(0));
+						removeLinks.push(link);
+					}
+				}
+			}
+		}
+		
+		for (link in removeLinks) {
+			this.deleteLink(link);
+		}
+		
+		var markedLinks = new Array<Link>();
+		
+		// case : both endpoints of a link are in the same connection then the link is marked
+		for (link in this.linkArray) {
+			if (link.get_endpoint(0).getConnection() == link.get_endpoint(1).getConnection()) {
+				markedLinks.push(link);
+			}
+		}
+		
+		// delete all marked links
+		for (link in markedLinks) {
+			this.deleteLink(link);
+		}
+		markedLinks = new Array<Link>();
+		
+		// For all remaining links x
+		// 		If x is not marked for deletion
+		// 			For all links y connected to x.endpoint(0) other than x
+		// 				If the other end of y connects to x.endpoint(0)
+		// 					Mark y for deletion
+		for (link in this.linkArray) {
+			var otherLinks = this.linkArray;
+			otherLinks.remove(link);
```

Comment:
Don't to that.  This will remove link from this.linkArray.  Instead do a nested iteration on this.linkArray and skip the case where the links are the same.